servicetalk-http-api/src/test/java/io/servicetalk/http/api/DefaultHttpExecutionStrategyTest.java

```diff
@@ -214,11 +217,26 @@ void wrapServiceThatWasNotOffloaded(final Params params) throws Exception {
                 new TestHttpServiceContext(INSTANCE, respFactory,
                         // Use noOffloadsStrategy() for the ctx to indicate that there was no offloading before.
                         // So, the difference function inside #offloadService will return the tested strategy.
-                        new ExecutionContextToHttpExecutionContext(contextRule,
-                                noOffloadsStrategy()));
-        analyzer.instrumentedResponseForServer(svc.handle(ctx, req, ctx.streamingResponseFactory()))
-                .flatMapPublisher(StreamingHttpResponse::payloadBody)
-            .toFuture().get();
+                        new ExecutionContextToHttpExecutionContext(contextRule, noOffloadsStrategy()));
+        CountDownLatch awaitComplete = new CountDownLatch(1);
+        Runnable runHandle = () -> {
+            try {
+                analyzer.instrumentedResponseForServer(svc.handle(ctx, req, ctx.streamingResponseFactory()))
+                        .flatMapPublisher(StreamingHttpResponse::payloadBody)
+                    .toFuture().get();
+            } catch (InterruptedException | ExecutionException e) {
+                fail("Unexepected exception", e);
+            } finally {
+              awaitComplete.countDown();
+            }
+        };
+        if (params.offloadSend || params.offloadReceiveMeta || params.offloadReceiveData) {
+            NettyIoExecutor ioExecutor = (NettyIoExecutor) contextRule.ioExecutor();
+            ioExecutor.asExecutor().execute(runHandle);
```

Comment:
you can remove the `awaitComplete` latch if you change this line to `ioExecutor.asExecutor().submit(runHandle).toFuture().get();`